src/Cake.Common/Build/TFBuild/Data/TFBuildPublishTestResultsData.cs

```diff
@@ -0,0 +1,93 @@
+using Cake.Core;
+using Cake.Core.IO;
+
+namespace Cake.Common.Build.TFBuild.Data
+{
+    /// <summary>
+    ///  Provides TF Build agent info for publishing test results
+    /// </summary>
+    public class TFBuildPublishTestResultsData
+    {
+        /// <summary>
+        /// Gets or Sets the type test runner the results are formated in
+        /// </summary>
+        public TFTestRunnerType? TestRunner { get; set; }
+
+        /// <summary>
+        /// Gets or sets the list of Test Result files to publish.
+        /// </summary>
+        public string[] TestResultsFiles { get; set; }
+
+        /// <summary>
+        /// Gets or Sets whether to merge all Test Result Files into one run
+        /// </summary>
+        public bool? MergeTestResults { get; set; }
+
+        /// <summary>
+        /// Gets or Sets the Platform for which the tests were run on
+        /// </summary>
+        public string Platform { get; set; }
+
+        /// <summary>
+        /// Gets or Sets the configuration for which the tests were run on
+        /// </summary>
+        public string Configuration { get; set; }
+
+        /// <summary>
+        /// Gets or Sets a name for the Test Run.
+        /// </summary>
+        public string TestRunTitle { get; set; }
+
+        /// <summary>
+        /// Gets or Sets whether to opt in/out of publishing test run level attachments
+        /// </summary>
+        public bool? PublishRunAttachments { get; set; }
+
+        internal Dictionary<string, string> GetProperties(ICakeEnvironment environment)
+        {
+            if (environment == null)
+            {
+                throw new ArgumentNullException(nameof(environment));
+            }
+
+            var properties = new Dictionary<string, string>();
+
+            if (TestRunner.HasValue)
+            {
+                properties.Add("type", TestRunner.Value.ToString());
+            }
+            if (MergeTestResults.HasValue)
+            {
+                properties.Add("mergeResults", MergeTestResults.ToString().ToLowerInvariant());
+            }
+            if (!string.IsNullOrWhiteSpace(Platform))
+            {
+                properties.Add("platform", Platform);
+            }
+            if (!string.IsNullOrWhiteSpace(Configuration))
+            {
+                properties.Add("config", Configuration);
+            }
+            if (!string.IsNullOrWhiteSpace(TestRunTitle))
+            {
+                properties.Add("runTitle", $"'{TestRunTitle}'");
+            }
+            if (PublishRunAttachments.HasValue)
+            {
+                properties.Add("publishRunAttachments", PublishRunAttachments.ToString().ToLowerInvariant());
+            }
+            if (TestResultsFiles != null && TestResultsFiles.Any())
+            {
+                properties.Add("resultFiles", string.Join(";", TestResultsFiles.Select(filePath => new FilePath(filePath).MakeAbsolute(environment).FullPath.Replace("/", "\\"))));
```

Comment:
@louisfischer Trying to educate myself- how did you find out that it should be semicolon-separated?

Reply:
@jnm2 thanks for that. That is a typo on my part. It should actually be a comma. Turning On System.Debug Variable shows the Logging Commands used by the various tasks built into TFS/VSTS.  

The snippet below (Publish Test Results)  just shows that even the canned TFS tasks use the Logging Commands.

```cmd
adjustedPattern: 'E:\tfsagents\agent4\_work\1\s\test/results/*.xml'
2 matches
2 final results
Reading test results from file 'E:\tfsagents\agent4\_work\1\s\test\results\TestResults.xml'
Reading test results from file 'E:\tfsagents\agent4\_work\1\s\test\results\TestResults2.xml'
Processed: ##vso[results.publish type=NUnit;mergeResults=false;runTitle=17179;publishRunAttachments=true;resultFiles=E:\tfsagents\agent4\_work\1\s\test\results\TestResults.xml,E:\tfsagents\agent4\_work\1\s\test\results\TestResults2.xml;]
task result: Succeeded
Processed: ##vso[task.complete result=Succeeded;]
```
I will be updating the PR for with the right separator.

 I am not sure why the PublishTestResults and PublishCodeCoverage results are not on the [Logging Commands](https://github.com/Microsoft/vsts-tasks/blob/master/docs/authoring/commands.md) documentation page.

@gep13 I read your question to the TFS team that posed a question about this. Did you get any response from them? I was wondering if they would take a pull request to update the documentation. I would be glad to update the docs. I am just not sure if a similar request has already been submitted.